### PR TITLE
#164829324 Implement User Login

### PR DESCRIPTION
**Pull request by MainaKamau92**
### What does this PR do?
* Implements front end login functionality for the authors haven web app.

### Description of Tasks to be completed?
* Give the ability to login for registered users
* Give descriptive validation error messages 

### How should this be manually tested?
* Clone the project: git clone https://github.com/andela/ah-the-answer-frontend
* cd into the project folder
* checkout to branch ft-implement-user-login-164829324
* Install dependencies: `npm install`
* Start server `npm start`
* Navigate to:
  - Local:  http://localhost:3000/
  - On Your Network:  http://10.10.182.30:3000/

### Screenshots
* Loaded login page
Screenshot 2019-05-08 at 10 18 18

* On unsuccessful login attempt
Screenshot 2019-05-08 at 14 23 44

### What are the relevant Pivotal Tracker Stories?
[164829325](https://www.pivotaltracker.com/story/show/164829325)

### CODE REVIEW

**Review by kihuha — COMMENTED** [review]
Please add the folder '.vscode/settings.json' to your .gitignore

File: src/store/actions/authActions.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import axios from 'axios';
+
+export const signInUser = userData => async (dispatch) => {
+  const data = {
+    user: {
+      ...userData,
+    },
+  };
+  await axios
+    .post('http://127.0.0.1:8000/api/users/login/', data)
```
Comment (kihuha) [inline src/store/actions/authActions.js]: I think the url should be the Heroku app url instead of local host

**Review by kihuha — DISMISSED** [review]

**Review by kihuha — DISMISSED** [review]

**Review by kihuha — DISMISSED** [review]

**Review by Raywire — DISMISSED** [review]
Login works as expected.

**Review by kirega — DISMISSED** [review]
The functionality is good, works as I expected

**Comment by kihuha** [comment]
I don't think the navbar is required in the login page. Just have a button at the bottom in case a user wants to go back home.

**Review by kihuha — DISMISSED** [review]
Looks good

**Review by Raywire — DISMISSED** [review]
Great work on implementing the login feature.

**Comment by WNjihia** [comment]
@MainaKamau92 there are a couple of merge conflicts. Kindly look into them

**Review by Raywire — DISMISSED** [review]

**Review by kihuha — COMMENTED** [review]
The login Route has not been included in app.js

**Review by kihuha — APPROVED** [review]

**Review by Raywire — APPROVED** [review]
Looks good.

**Review by WNjihia — APPROVED** [review]

**Pull request merged**